frame/support/src/traits/metadata.rs

```diff
@@ -224,6 +224,16 @@ impl PartialOrd<u16> for StorageVersion {
 	}
 }
 
+/// Special marker struct if no storage version is set for a pallet.
+///
+/// If you (the reader) end up here, it probably means that you tried to compare
+/// [`GetStorageVersion::on_chain_storage_version`] against
+/// [`GetStorageVersion::current_storage_version`]. This basically means that the
+/// [`storage_version`](crate::pallet_macros::storage_version) is missing in the pallet where the
+/// mentioned functions are being called.
+#[derive(Debug, Default)]
+pub struct NoStorageVersionSet;
```

Comment:
Why do we need this instead of using `Option<StorageVersion>` as return value on `on_chain_storage_version`?  
Backwards compatibility maybe…

Reply:
I also first had this, but then we would always need to compare with this option and I didn't liked the approach. When there is a version set we then either need to use `expect`/`unwrap` or some extra check code. I think this way it is easier to use, because most of the time we will have a version set when we call this function.

Reply:
Yea…always unwrapping is also stupid. Then this is okay.